File: .skill-build/Presentation/G1/python/Lorenz_Full_Info_H_Lin_plot.py
"""Reproduce the G1 EKF/KalmanNet figure without a shared MATLAB workspace."""
import argparse
from pathlib import Path
import numpy as np
import matplotlib.pyplot as plt
try:
    from .Lorenz_Full_Info_H_Lin_data import load_data
    from .myLabel import myLabel
    from . import style as s
except ImportError:
    from Lorenz_Full_Info_H_Lin_data import load_data
    from myLabel import myLabel
    import style as s


def plot_results(data=None, ax=None):
    """Return (figure, axes). Plot only the first two ratios, as in MATLAB."""
    data = load_data() if data is None else data
    if ax is None:
        _, ax = plt.subplots(figsize=(24, 14))
    x = -np.asarray(data['r2_arr_dB'])
    for q_idx in range(2):
        ratio_db = 20 * np.log10(data['q_arr'][q_idx])
        label = s.q_str + f' {ratio_db:g} ' + s.db_str + s.del_str
        for key, prefix, name in [('MSE', 'KF', s.EKF_str),
                                   ('MSE_KNET', 'KNET', s.KNET_str)]:
            ax.plot(x, data[key][:, q_idx],
                    color=getattr(s, prefix + '_COLOR')[q_idx],
                    linestyle=getattr(s, prefix + '_LINE')[q_idx],
                    linewidth=getattr(s, prefix + '_LINE_WIDTH'),
                    marker=getattr(s, prefix + '_MARKER')[q_idx],
                    markersize=getattr(s, prefix + '_MARKER_SIZE'),
                    markerfacecolor='none', markeredgecolor='k', label=label + name)
    ax.plot(x, data['r2_arr_dB'], color='r', linestyle='-.',
            linewidth=s.KNET_LINE_WIDTH, label=s.BL_str)
    ax.set_xlim(0, 40)
    ax.set_ylim(-51, 0)
    ax.tick_params(axis='both', labelsize=30)
    ax.grid(True)
    ax.legend(fontsize=35, loc='upper right', framealpha=1)
    myLabel(ax)
    ax.figure.tight_layout()
    return ax.figure, ax


def main():
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument('--save', type=Path, help='Output PNG, PDF, SVG, or EPS path')
    parser.add_argument('--no-show', action='store_true', help='Render without a GUI')
    args = parser.parse_args()
    if args.no_show:
        plt.switch_backend('Agg')
    fig, _ = plot_results()
    if args.save:
        args.save.parent.mkdir(parents=True, exist_ok=True)
        fig.savefig(str(args.save), dpi=160, bbox_inches='tight')
        print(f'Saved {args.save.resolve()}')
    if not args.no_show:
        plt.show()
    else:
        plt.close(fig)


if __name__ == '__main__':
    main()

File: .skill-build/Presentation/G1/Lorenz_Full_Info_H_Lin_plot.m
close all

style


J_work_EKF = [1] ;

J_work_KNET = [1] ;

J_mdl_arr = [5] ;

%%%%%%%%%%%%%%%%%%%%
%%% Define Legend %%
%%%%%%%%%%%%%%%%%%%%

J_num_EKF = num2str(J_mdl_arr(J_work_EKF)) ;
J_num_KNET = num2str(J_mdl_arr(J_work_KNET)) ;

% myLegend = [
%     q_str + " 0" + db_str + del_str +  EKF_str + del2_str + J_mdl_str + space_str + J_num_EKF(1),
%     q_str + " 0" + db_str + del_str +  KNET_str + del2_str + J_mdl_str + space_str + J_num_EKF(1),
%     q_str + " -20" + db_str + del_str +  EKF_str + del2_str + J_mdl_str + space_str + J_num_EKF(1),
%     q_str + " -20" + db_str + del_str +  KNET_str + del2_str + J_mdl_str + space_str + J_num_EKF(1),
%     BL_str
% ] ;

myLegend = [
    q_str + " 0" + db_str + del_str +  EKF_str
    q_str + " 0" + db_str + del_str +  KNET_str
    q_str + " -20" + db_str + del_str +  EKF_str
    q_str + " -20" + db_str + del_str +  KNET_str
    BL_str
] ;

Data = MSE ;
Data_KNET = MSE_KNET ;


figure

set(gca,'FontSize',80)


for qIdx=1:1:2

%%%%%%%%%%%%
%%% Plot %%%
%%%%%%%%%%%%'

    plot(-r2_arr_dB, Data(:, qIdx), 'Color', KF_COLOR(qIdx), 'LineStyle', KF_LINE(qIdx), 'LineWidth', KF_LINE_WIDTH, 'Marker', KF_MARKER(qIdx), 'MarkerSize', KF_MARKER_SIZE, 'MarkerEdgeColor', 'k') ;

    hold on


    plot(-r2_arr_dB, Data_KNET(:, qIdx), 'Color', KNET_COLOR(qIdx), 'LineStyle', KNET_LINE(qIdx), 'LineWidth', KNET_LINE_WIDTH, 'Marker', KNET_MARKER(qIdx), 'MarkerSize', KNET_MARKER_SIZE, 'MarkerEdgeColor', 'k') ;

    hold on
 
end

plot(-r2_arr_dB, r2_arr_dB, 'Color', 'r', 'LineStyle', '-.', 'LineWidth', KNET_LINE_WIDTH, 'MarkerEdgeColor', 'k') ;

    %%%%%%%%%%%%%%%%%%%%%
    %%% Figure Config %%%
    %%%%%%%%%%%%%%%%%%%%%

    
    xlim([0, 40])
    ylim([-51, 0])

    ax = gca ;
    ax.XAxis.FontSize = 30;
    ax.YAxis.FontSize = 30;

    grid

    %%%%%%%%%%%%%%
    %%% Legend %%%
    %%%%%%%%%%%%%%
    lgd = legend(myLegend, 'Interpreter', 'latex');
    lgd.FontSize = 35;
    
    
    %%%%%%%%%%%%%
    %%% Label %%%
    %%%%%%%%%%%%%
    myLabel


    




